src/api/resolvers/tokenResolvers.js

```diff
@@ -0,0 +1,93 @@
+import getWeb3, { getAccount, getTokenBySymbol } from '../web3'
+import { Token } from '@wearekickback/contracts'
+import { txHelper, EMPTY_ADDRESS } from '../utils'
+export const defaults = {}
+const abi = Token.abi
+let token
+
+const resolvers = {
+  Query: {
+    async getTokenBySymbol(_, { symbol }) {
+      const address = await getTokenBySymbol(symbol)
+      return {
+        address,
+        __typename: 'Token'
+      }
+    },
+    async getTokenAllowance(_, { tokenAddress, partyAddress }) {
+      const obj = {
+        account,
+        __typename: 'Token'
+      }
+      const web3 = await getWeb3()
+      const account = await getAccount()
+
+      const { methods: contract } = new web3.eth.Contract(abi, tokenAddress)
+      try {
+        const account = await getAccount()
+        const allowance = await contract.allowance(account, partyAddress).call()
+        return {
+          ...obj,
+          allowance
+        }
+      } catch (err) {
+        console.log('Failed to fetch tokenAllowance', err)
+        return {
+          ...obj,
+          allowance: null
+        }
+      }
+    },
+    async getToken(_, { tokenAddress }) {
+      if (token) {
+        return token
+      }
+
+      if (!tokenAddress || tokenAddress === EMPTY_ADDRESS) {
+        return {
```

Comment:
Given I am not extracting into a contract file, I just created `isEmptyAddress` under `utils`

Reply:
Misunderstanding on this one. I didn't mean the isEmpty part, but the object for Ether could be a const somewhere just in case we needed it somewhere else